yt_dlp/extractor/tv2hu.py

```diff
@@ -2,61 +2,117 @@
+            entries = [self.url_result(f'https://tv2play.hu/szalag/{ribbon_id}',
+                                       ie=TV2HuSeriesIE.ie_key(), video_id=ribbon_id) for ribbon_id in ribbon_ids]
+            return self.playlist_result(entries, playlist_id=id)
+        elif json_data['contentType'] != 'video':
+            raise ExtractorError('Incorrect URL.', expected=True)
+
+        video_id = str(json_data['id'])
+        title = json_data['title']
+        series = json_data.get('seriesTitle')
+        duration = json_data.get('length')
+        description = json_data.get('description')
+        thumbnail = 'https://tv2play.hu' + json_data.get('thumbnailUrl')
+        release_date = json_data.get('uploadedAt').replace('.', '')
+        player_id = json_data.get('playerId')
+        series_json = json_data.get('seriesInfo', {})
+        episode_number = series_json.get('episodeNr')
+        season_number = series_json.get('seasonNr')
+
+        video_json_url = self._download_json(f'https://tv2play.hu/api/streaming-url?playerId={player_id}', video_id)['url']
+        video_json = self._download_json(video_json_url, video_id)
+        m3u8_url = traverse_obj(video_json, ('bitrates', 'hls')).replace('//', 'https://')
+        formats, subtitles = self._extract_m3u8_formats_and_subtitles(m3u8_url, video_id)
         self._sort_formats(formats)
 
         return {
             'id': video_id,
-            'title': self._og_search_title(webpage).strip(),
-            'thumbnail': self._og_search_thumbnail(webpage),
-            'upload_date': self._search_regex(
-                r'/vod/(\d{8})/', json_url, 'upload_date', default=None),
+            'title': title,
+            'series': series,
+            'duration': duration,
+            'description': description,
+            'thumbnail': thumbnail,
+            'release_date': release_date,
+            'season_number': season_number,
+            'episode_number': episode_number,
             'formats': formats,
+            'subtitles': subtitles,
         }
+
+
+class TV2HuSeriesIE(InfoExtractor):
+    IE_NAME = 'tv2playseries.hu'
+    _VALID_URL = r'https?://(?:www\.)?tv2play\.hu/szalag/(?P<id>[^#&?]+)'
+
+    _TESTS = [{
+        'url': 'https://tv2play.hu/szalag/59?rendezes=nepszeruseg',
+        'playlist_mincount': 284,
+        'info_dict': {
+            'id': '59',
+        }
+    }]
+
+    def _real_extract(self, url):
+        id = self._match_id(url)
+        json_data = self._download_json(f'https://tv2play.hu/api/ribbons/{id}/0?size=100000', id)
+        entries = []
+        for card in json_data.get('cards', []):
+            video_id = card.get('slug')
+            if video_id:
+                entries.append(self.url_result(f'https://tv2play.hu/{video_id}',
+                                               ie=TV2HuIE.ie_key(), video_id=video_id))
+
+        return self.playlist_result(entries, playlist_id=id)
```

Comment:
see `playlist_from_matches`

Reply:
it doesn't support `video_id` afaict

Reply:
gotcha. still, making an empty list and appending is considered bad practice in python and should be avoided if possible
```suggestion
        entries = [self.url_result(f'https://tv2play.hu/%s' % card['slug'], ie=TV2HuIE.ie_key(), video_id=card['slug'])
                   for card in json_data.get('cards', []) if 'slug' in card]

        return self.playlist_result(entries, playlist_id=id)
```

Reply:
> making an empty list and appending is considered bad practice in python 

Isn't this done in every other extractor?  -_-

Reply:
ig. doesnt mean we cant do better 🤷

Reply:
it will give incorrect result on `'slug': None`
```
Python 3.9.5 (default, Jun  4 2021, 12:28:51) 
[GCC 7.5.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> a = {'c': None}
>>> 'c' in a
True
>>> 
```

Reply:
does that happen?

Reply:
we can also just use `if card.get('slug')` imo. but if dont wanna change this, its fine, no big deal

Reply:
hypothetically yes,
The sites uses `null` when a field is not there which is  parsed as `None` in python.
I haven't found a concrete example of this in `card` tho.

Reply:
I personally like generators better aswell. But thing is if we use generator here, we'll be calling card['slug’] or .get thrice. Once for url, once for video id and once for if statement. 
That's the reason for not using generator.